### PR TITLE
Next version

### LINKED ISSUE #13: `wfs_wcs` vignette fails to knit
Vignette "wfs_wcs.Rmd" fails to build: the Belgian WFS server seems to be gone.

- https://eservices.minfin.fgov.be/arcgis/services/R2C/Regions/MapServer/WFSServer

I could not find an equivalent server (geo.be seems to be a clue, but does not hold map services).

### PR DESCRIPTION
**Pull request by hansvancalster**
Branch to collect contributions for new version of the package.

Closes #13

### CODE REVIEW

**Comment by hansvancalster** [comment]
@falkmielke I will ask you to review this branch, but it is more a pro-forma question since most of the stuff here is already reviewed in the other branch. The only substantial addition is a small change in the unpack_mht function to also detect geoTIFF with different type of byte order.

@florisvdh would you like to review too? Or do you perhaps have something in mind to contribute? This would be a good time.

**Review by falkmielke — APPROVED** [review]
I could successfully install the `next-version` branch locally, vignettes build as expected, outcomes look fine. I have also looked at all code changes, seems good. Thank you for bringing together the new version!

For the archives: I have kept notes to better test the geoTIFF extraction (not tested now) as examples roll over my desk; I assume it worked for your cases. I also added to my todo list to create unit tests for the different wcs services, where we query known outcomes. In particular, you mentioned a strict CRS 31370 requirement, which I will double check once I understand more about that. 
Since the package works, and we practically only use 31370, these extras are not urgent. But let me know if we should keep track via an issue.

File: vignettes/spatial_dhmv_query.Rmd [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+|------------------- |--------------------------------- |
+| service             | WCS                               |
+| version             | 2.0.1                             |
+| coverageId          | DHMVII_DTM_1m |
+| subset              | x(148599,148601)                  |
+| subset              | y(208899,208901)                  |
+| format              | image/tiff                        |
+| request             | GetCoverage                       |
+
+You can selectively shuffle and adjust components, generalizing the working query and learning about the WCS API. Which feels a bit like [fuzzing](https://en.wikipedia.org/wiki/Fuzzing) the `geo.api`.
```
Comment (florisvdh) [inline vignettes/spatial_dhmv_query.Rmd:255]: 'geo.api' confused me a bit here (hence so does the geo.api subdomain). I was confusing with the recent [OGC API](https://ogcapi.ogc.org/), i.e. the successor of OGC Web Service Standards.  IIUC, OGC API has dedicated server/client implementations in Java and Python, called the OGC "[GeoAPI](https://www.ogc.org/publications/standard/geoapi/)" standard, and is still evolving. The OGC API offers modern Web API approaches compared to the (legacy!) OGC Web Service Standards (which will probably remain in place for a long time).

Yet we shouldn't confuse with OGC API / GeoAPI since we're dealing with web services, not OGC API.

BTW, interacting with OGC API is provided by GDAL ([vector](https://gdal.org/en/latest/drivers/vector/oapif.html) / [raster](https://gdal.org/en/latest/drivers/raster/ogcapi.html)), which is the obvious gateway for R packages like {sf} and {terra}, as these already interface GDAL.

(Also, the webservice standards are supported by GDAL, hence should be able to use with GDAL-interfacing R packages)

The term 'geo.api' pops up a couple of times in the vignette so maybe you can improve or replace it (maybe 'WCS request' or something).

File: R/get_coverage_wcs.R [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #' from which it is read with `terra::rast()` - if needed reprojected -
 #' and returned as a `SpatRaster`object
```
Comment (florisvdh) [inline R/get_coverage_wcs.R:5]:
```suggestion
#' and returned as a `SpatRaster` object
```

(This is actually not part of this PR)

File: R/get_coverage_wcs.R [hunk 3]
```diff
@@ -20,7 +20,8 @@
 #'   - `"omw"`: orthophotomosaic winter images Flanders
 #'   - `"dtm"`: digital terrain model Flanders
 #'   - `"dsm"`: digital surface model Flanders
-#' See [metadata Vlaanderen](https://metadata.vlaanderen.be/metadatacenter/srv/dut/catalog.search#/search?keyword=OGC:WCS) for more information # nolint: line_length_linter.
+#'   - `"dhmv"`: digital elevation model Flanders (contains dtm and dsm data)
+#' See [metadata Vlaanderen](https://metadata.vlaanderen.be/srv/eng/catalog.search#/search?any=WCS) for more information # nolint: line_length_linter.
```
Comment (florisvdh) [inline R/get_coverage_wcs.R:24]: The `# nolint: line_length_linter` string appears as-is in the resulting Rd file... I don't know whether roxygen comments have a way to support this, but the current result doesn't look so nice :confused:.

**Review by florisvdh — COMMENTED** [review]
This review (more or less) adds to the review of PR #12.

@falkmielke I've read the spatial_dhmv_query vignette. Pleasant reading, relatively easy to follow, well done! :sun_with_face: 

I locally tested the `get_coverage_wcs()` example in the vignette, with success!

Only minor comments added.

A minor general one is my advice to put each sentence of a paragraph in another line. Hugo will stitch those lines into one paragraph in HTML, as would Pandoc. It minimizes git diffs and it will also make review comments easier to interpret afterwards. No need to implement it here as far as I'm concerned, just saying!

Don't forget to rebuild the documentation (Rd files) after making updates in the roxygen comment section.

File: vignettes/spatial_dhmv_query.Rmd [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+```{r}
+caps <- wcs$getCapabilities()
+```
+
+This would grab the same info as you can find in this xml file: <https://geo.api.vlaanderen.be/DHMV/wcs?service=WCS&request=getcapabilities>
+
+Note that the *capabilities* xml holds a lot of valuable information:
+
+-   First and foremost, search it for `CoverageID`; in this case, I was interested in `DHMVII_DTM_1m`: grid coverage of the LIDAR elevation data for Flanders.
+-   We can also find that the Coordinate Reference System (CRS) `crsSupported` is EPSG:31370, aka. [Belgian Lambert 72](https://epsg.io/31370). Always good to know.
```
[the remark this answers is not in the document]
Comment (falkmielke) [reply vignettes/spatial_dhmv_query.Rmd:93]: Thank you for the pointers, which I have read and stored. 
I will adjust and include the dynamic CRS naming.

File: vignettes/spatial_dhmv_query.Rmd (hunk 1 above)
Comment (falkmielke) [reply vignettes/spatial_dhmv_query.Rmd:255]: Interesting and good to know! I have replaced all "API" by WCS or interfaces, except when writing about the `geo.api.vlaanderen.be`.

File: R/get_coverage_wcs.R (hunk 3 above)
Comment (falkmielke) [reply R/get_coverage_wcs.R:24]: I have rearranged that comment.

File: vignettes/spatial_dhmv_query.Rmd [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+
+<a id="org575ca41"></a>
+
+# *Post Mortem*
+
+The value of this vignette seems to be limited: in essence, it provides the specific query function arguments for elevation model queries with the `geo.api` of the Flemish digital services. Quite a niche use case.
+
+However, this is also a story of tracing errors in existing tools, and scraping other tools for hints.
+
+Can we actually trace the error back to a source? It was pure co-incidence that QGIS provided a working link, or are there general strategies?
```
[the remark this answers is not in the document]
Comment (falkmielke) [reply vignettes/spatial_dhmv_query.Rmd:463]: Thank you for confirming; I took the liberty of copying most of this comment into the vignette, it seems worth re-iterating the praise of QGIS.

File: vignettes/spatial_dhmv_query.Rmd (hunk 1 above)
Comment (florisvdh) [reply vignettes/spatial_dhmv_query.Rmd]: Thanks!
